rust/sdk-processor/src/steps/common/parquet_buffer_step.rs

```
@@ -0,0 +1,387 @@
+{
+    type Input = HashMap<ParquetTypeEnum, ParquetTypeStructs>;
+    type Output = HashMap<ParquetTypeEnum, TransactionMetadata>;
+    type RunType = PollableAsyncRunType;
+
+    /// Processes incoming `TransactionContext` data by appending it to the appropriate buffers.
+    ///
+    /// If any buffer exceeds the maximum size, its contents are uploaded.
+    /// Returns metadata information on what was uploaded.
+    async fn process(
+        &mut self,
+        item: TransactionContext<Self::Input>,
+    ) -> Result<Option<TransactionContext<Self::Output>>, ProcessorError> {
+        debug!("Starting process for {} data items", item.data.len());
+
+        let mut upload_metadata_map = HashMap::new();
+        for (parquet_type, parquet_data) in item.data {
+            self.handle_buffer_append(
+                parquet_type,
+                parquet_data,
+                &item.metadata,
+                &mut upload_metadata_map,
+            )
+            .await?;
+        }
+
+        if !upload_metadata_map.is_empty() {
+            return Ok(Some(TransactionContext {
+                data: upload_metadata_map,
+                metadata: item.metadata, // cur_batch_metadata,
+            }));
+        }
+        Ok(None)
+    }
+
+    async fn cleanup(
+        &mut self,
+    ) -> Result<Option<Vec<TransactionContext<Self::Output>>>, ProcessorError> {
+        let mut metadata_map = HashMap::new();
+        debug!("Starting cleanup: uploading all remaining buffers.");
+        for (parquet_type, mut buffer) in self.internal_buffers.drain() {
+            if buffer.buffer_size_bytes > 0 {
+                let struct_buffer = std::mem::replace(
+                    &mut buffer.buffer,
+                    ParquetTypeStructs::default_for_type(&parquet_type),
+                );
+
+                self.buffer_uploader.handle_buffer(struct_buffer).await?;
+
+                if let Some(buffer_metadata) = &mut buffer.current_batch_metadata {
```

if it's None, can we return an error?